### DISCUSSION TITLE
Vue Warnings using Plugin with Options API?

### DISCUSSION THREAD

**Discussion by jamiedoris** [ankurk91/vue-loading-overlay#118] [Q&A]
I'm migrating a project from Vue 2 to Vue 3, and I'm seeing a warning that reads as follows:

`[Vue warn]: onMounted is called when there is no active component instance to be associated with. Lifecycle injection APIs can only be used during execution of setup(). If you are using async setup(), make sure to register lifecycle hooks before the first await statement. `

I believe I've narrowed this down to something inside vue-loading-overlay. I'm using version 6.0.3 with the latest Vue, and here's what my implementation looks like:

_main.js:_
```
import { LoadingPlugin } from 'vue-loading-overlay';
import 'vue-loading-overlay/dist/css/index.css';
```

_Component method:_
```
    doDispatch (apiParams) {
      const loading = this.$loading.show()
      this.$store.dispatch('ACTION_NAME', apiParams).then(() => {
        loading.hide()
      })
    },
```

The loader works fine, but I'd like to get rid of the warning.
Anyone else seeing this?
Any ideas?

**Comment by ankurk91** [comment]
This package is still using options api under the hood.

I have not used vue 3 in any of my real project yet.

All organizations are struggling with migration to vue 3.

I will try to look into it, but can't promise

**Reply by jamiedoris** [reply]
It occurs to me that my method is part of a mixin, and I'm using the migration build. Maybe it's because of the mixin. I'll see if I can rearch a bit, which I have to do eventually anyway. Thanks for the quick reply.
